### PR TITLE
Cleanup FairAlignmentHandler

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
* Apply some clang-tidy cleanups
* Use TRangeDynCast in RecomputePhysicalAssmbBbox

@PALoizeau as you authored #1244, could you take a look at this?

### CODE REVIEW

File: alignment/FairAlignmentHandler.cxx
Comment (fuhlig1) [inline alignment/FairAlignmentHandler.cxx:105]: Looks good. Did not know this.

Comment (PALoizeau) [reply alignment/FairAlignmentHandler.cxx:105]: Same for me, nice to learn about this functionality, it may be useful later (even we use less ROOT style containers/collection with direct access recently).

**Comment by PALoizeau** [comment]
I am not the original author of most of these blocks, but I think this looks good and most of the changes would be what I would do if asked to cleanup/refresh a class. So :+1:  from my side

**Comment by ChristianTackeGSI** [comment]
@karabowi Do we want this in 19.0? Then you or @fuhlig1 might just press the merge button?

If we want it in 19.2, then we should set the milestone and wait until after the release of 19.0?

**Comment by fuhlig1** [comment]
In my opinion it would be good to have it even in 18.8.

**Review by fuhlig1 — APPROVED** [review]

**Pull request merged**
